File: exemplos/aula_250513/pagamento/pagamento.py
from abc import ABC, abstractmethod

class Pagamento(ABC):
    def __init__(self, valor):
        self.valor = valor

    @abstractmethod
    def processarPagamento(self):
        pass
    
class CartaoCredito(Pagamento):
    def __init__(self, valor, numero_cartao):
        super().__init__(valor)
        self.numero_cartao = numero_cartao

    def processarPagamento(self):
        print(f'Pagamento de {self.valor} processado via Cartão de Crédito: {self.numero_cartao}')
        
class Boleto(Pagamento):
    def __init__(self, valor, codigo_barras):
        super().__init__(valor)
        self.codigo_barras = codigo_barras

    def processarPagamento(self):
        print(f'Pagamento de {self.valor} processado via Boleto: {self.codigo_barras}')
        
class Pix(Pagamento):
    def __init__(self, valor, chave_pix):
        super().__init__(valor)
        self.chave_pix = chave_pix

    def processarPagamento(self):
        print(f'Pagamento de {self.valor} processado via Pix: {self.chave_pix}')
        
# Exemplo de uso
pagamento_cartao = CartaoCredito(100.0, '1234-5678-9012-3456')
pagamento_cartao.processarPagamento()

File: exemplos/aula_250513/pagamento/Pagamento.java
package exemplos.aula_250513.pagamento;

abstract class Pagamento {
    protected double valor;

    public abstract void processar();
}

class PagamentoCartao extends Pagamento {
    private String numeroCartao;
    @Override
    public void processar() {
        System.out.println("Processando pagamento com cartão: " + valor);
    }
}

class PagamentoBoleto extends Pagamento {
    private String codigoBoleto;
    @Override
    public void processar() {
        System.out.println("Processando pagamento com boleto: " + valor);
    }
}

class PagamentoPix extends Pagamento {
    private String chavePix;
    @Override
    public void processar() {
        System.out.println("Processando pagamento com Pix: " + valor);
    }
}

class App {
    public static void main(String[] args) {
        Pagamento pagamento = new PagamentoCartao();
        pagamento.valor = 100.0;
        pagamento.processar();

        pagamento = new PagamentoBoleto();
        pagamento.valor = 200.0;
        pagamento.processar();

        pagamento = new PagamentoPix();
        pagamento.valor = 300.0;
        pagamento.processar();
    }
}